website_local_fonts/static/src/js/snippets.options.js

```diff
@@ -0,0 +1,270 @@
+            _.times(nbFonts, (fontNb) => {
+                const realFontNb = fontNb + 1;
+                const fontEl = document.createElement("we-button");
+                fontEl.classList.add(`o_we_option_font_${realFontNb}`);
+                fontEl.dataset.variable = variable;
+                fontEl.dataset[methodName] = weUtils.getCSSVariableValue(
+                    `font-number-${realFontNb}`,
+                    style
+                );
+                fontEl.dataset.font = realFontNb;
+                fontEls.push(fontEl);
+                this.menuEl.appendChild(fontEl);
+            });
+            if (this.localFonts.length) {
+                const localFontsEls = fontEls.splice(-this.localFonts.length);
+                localFontsEls.forEach((el, index) => {
+                    $(el).append(
+                        core.qweb.render("website.delete_local_font_btn", {
+                            index: index,
+                        })
+                    );
+                });
+            }
+
+            if (this.googleLocalFonts.length) {
+                const googleLocalFontsEls = fontEls.splice(
+                    -this.googleLocalFonts.length
+                );
+                googleLocalFontsEls.forEach((el, index) => {
+                    $(el).append(
+                        core.qweb.render("website.delete_google_font_btn", {
+                            index: index,
+                            local: true,
+                        })
+                    );
+                });
+            }
+
+            if (this.googleFonts.length) {
+                const googleFontsEls = fontEls.splice(-this.googleFonts.length);
+                googleFontsEls.forEach((el, index) => {
+                    $(el).append(
+                        core.qweb.render("website.delete_google_font_btn", {
+                            index: index,
+                        })
+                    );
+                });
+            }
+
+            $(this.menuEl).append(
+                $(
+                    core.qweb.render("website.add_google_font_btn", {
+                        variable: variable,
+                    })
+                )
+            );
+            $(this.menuEl).append(
+                $(
+                    qweb.render("website.add_local_font_btn", {
+                        variable: variable,
+                    })
+                )
+            );
+        },
+
+        _onAddLocalFontClick: function (ev) {
+            const variable = $(ev.currentTarget).data("variable");
+            const dialog = new Dialog(this, {
+                title: _t("Add a Local Font"),
+                $content: $(qweb.render("website.dialog.addLocalFont")),
+                buttons: [
+                    {
+                        text: _t("Save & Reload"),
+                        classes: "btn-primary",
+                        click: async () => {
```

Comment:
I would have put this function inside the class for inheritance purposes. Non-blocking though.

Reply:
@tarteo , I have added the function in the class so that it can be inherited.Thanks for this suggestion !!!